### PR TITLE
Log exceptions with stack traces

### LINKED ISSUE #5070: InterruptedException can "escape" LS without being printed
This task is automatically imported from the old Task Issue Board and it was originally created by jaroslavtulach.
Original issue is [here](https://www.pivotaltracker.com/story/show/184349509).
<hr />

Investigating problems with `ConnectException` I realized that when `InterruptedException` is thrown from language server, it may remain unnoticed due to the way Scala deals with (checked) exceptions and classifies them to `NonFatal` and other.

The issue was discovered when searching for **reasons why the connection between IDE/engine breaks**. One of the hypothesis was: there is an exception in the engine, but we don't know about it as it is "swallowed". Then it doesn't appear in the logs and we cannot fix it, as we don't have any trace of it. This pivotal demonstrates such a situation can really happen - at least with `InterruptedException`.

To reproduce use following modification to simulate `InterruptedException` somewhere in the engine/language server code:
```
-- a/engine/language-server/src/main/scala/org/enso/languageserver/runtime/RuntimeKiller.scala
+++ b/engine/language-server/src/main/scala/org/enso/languageserver/runtime/RuntimeKiller.scala
@@ -68,9 +68,13 @@
         truffleContext,
         retryCount + 1
       )
+      if (true) {
+        throw new InterruptedException() 
+      } else {
       truffleContext.close()
       replyTo ! RuntimeGracefullyStopped
       context.stop(self)
+      }
     } catch {
       case NonFatal(ex) =>
         logger.error(
```
connect to the LS with IDE and then kill the IDE. The `InterruptedException` gets yielded, but it doesn't appear anywhere in the logs.

## Tasks:
- [ ] Make sure all exceptions are printed including `InterruptedException`

## Blockers:
[#184340534](https://www.pivotaltracker.com/story/show/#184340534) blocked

## Comments:
Isn't it logged in https://github.com/enso-org/enso/blob/develop/engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala#L239 (Hubert Plociniczak - Feb 1, 2023)
<hr />
The `InterruptedException` is not logged. I've got just:
```
[info] [2023-02-01T17:23:04.681Z] [org.enso.languageserver.protocol.binary.BinaryConnectionController] Connection closed [127.0.0.1:41502].
[info] [2023-02-01T17:23:04.683Z] [org.enso.languageserver.runtime.RuntimeKiller] Shutting down the runtime server [Actor[akka://language-server/user/runtime-connector#-183675620]].
[info] [2023-02-01T17:23:04.695Z] [org.enso.languageserver.runtime.RuntimeKiller] Shutting down the Truffle context [org.graalvm.polyglot.Context@75c86d65]. Attempt #1.
[internal-logger-error] One of the printers failed to write a message: java.lang.NullPointerException
[internal-logger-error] The dropped message was: [error] [2023-02-01T17:23:04.702Z] [akka.actor.OneForOneStrategy] null
[error] [2023-02-01T17:23:16.695Z] [org.enso.languageserver.boot.LanguageServerComponent] An error occurred during stopping server
[info] [2023-02-01T17:23:16.696Z] [org.enso.languageserver.boot.LanguageServerComponent] Terminated truffle context
[info] [2023-02-01T17:23:16.702Z] [org.enso.languageserver.boot.LanguageServerComponent] Terminated json connections
[info] [2023-02-01T17:23:16.703Z] [org.enso.languageserver.boot.LanguageServerComponent] Terminated binary connections
[info] [2023-02-01T17:23:16.717Z] [akka.actor.CoordinatedShutdown] Running CoordinatedShutdown with reason [ActorSystemTerminateReason]
[info] [2023-02-01T17:23:16.738Z] [org.enso.languageserver.boot.LanguageServerComponent] Terminated actor system
[info] [2023-02-01T17:23:16.846Z] [org.enso.languageserver.boot.LanguageServerComponent] Terminated main module
```
 (jaroslavtulach - Feb 1, 2023)
<hr />

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #5070

Changelog:
- Include the original exception to log expressions
- Enable logging of Akka Actors' lifecycle events on debug logging level
- Decrease the severity of interruption log messages because interruptions are part of the workflow. The computation can be interrupted at any time, and still be recomputed after. Warnings are just misleading in this case.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] ~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~

### CODE REVIEW

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala [hunk 1]
```diff
@@ -102,7 +102,7 @@ class MainModule(serverConfig: LanguageServerConfig, logLevel: LogLevel) {
       None,
       Some(serverConfig.computeExecutionContext)
     )
-  log.trace(s"Created ActorSystem $system.")
+  log.trace(s"Created ActorSystem [{}].", system)
```
Comment (hubertp) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala]:
```suggestion
  log.trace("Created ActorSystem [{}].", system)
```

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/LanguageServerComponent.scala [hunk 2]
```diff
@@ -39,17 +39,17 @@ class LanguageServerComponent(config: LanguageServerConfig, logLevel: LogLevel)
   override def start(): Future[ComponentStarted.type] = {
     logger.info("Starting Language Server...")
     val sampler = startSampling(config)
-    logger.debug(s"Started ${sampler.getClass.getName}.")
+    logger.debug(s"Started [{}].", sampler.getClass.getName)
```
Comment (hubertp) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/LanguageServerComponent.scala]:
```suggestion
    logger.debug("Started [{}].", sampler.getClass.getName)
```

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala [hunk 3]
```diff
@@ -165,15 +165,18 @@ class MainModule(serverConfig: LanguageServerConfig, logLevel: LogLevel) {
...
+              exception
             )
             new NoopEventsMonitor
         }
       case None =>
         new NoopEventsMonitor
     }
   log.trace(
-    s"Started runtime events monitor ${runtimeEventsMonitor.getClass.getName}."
+    s"Started runtime events monitor [{}].",
```
Comment (hubertp) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala]:
```suggestion
    "Started runtime events monitor [{}].",
```

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala [hunk 4]
```diff
@@ -165,15 +165,18 @@ class MainModule(serverConfig: LanguageServerConfig, logLevel: LogLevel) {
             monitor
           case Failure(exception) =>
             log.error(
-              s"Failed to create runtime events monitor for $path ($exception)."
+              s"Failed to create runtime events monitor for [{}].",
```
Comment (hubertp) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala]:
```suggestion
              "Failed to create runtime events monitor for [{}].",
```

File: engine/language-server/src/main/scala/org/enso/languageserver/filemanager/PathWatcher.scala [hunk 5]
```diff
@@ -116,14 +116,14 @@ final class PathWatcher(
       stopWatcher()
       restartCounter.inc()
       if (restartCounter.canRestart) {
-        logger.error(s"Restart on error#${restartCounter.count}", e)
+        logger.error(s"Restart #{} on error.", restartCounter.count, e)
```
Comment (hubertp) [inline engine/language-server/src/main/scala/org/enso/languageserver/filemanager/PathWatcher.scala]:
```suggestion
        logger.error("Restart #{} on error.", restartCounter.count, e)
```

File: engine/language-server/src/main/scala/org/enso/languageserver/libraries/LocalLibraryManager.scala [hunk 6]
```diff
@@ -205,8 +206,9 @@ class LocalLibraryManager(
       config.componentGroups match {
         case Left(error) =>
           logger.error(
-            s"Failed to parse library [$libraryName] component groups " +
-            s"(reason: ${error.message})."
+            s"Failed to parse library [{}] component groups.",
```
Comment (hubertp) [inline engine/language-server/src/main/scala/org/enso/languageserver/libraries/LocalLibraryManager.scala]:
```suggestion
            "Failed to parse library [{}] component groups.",
```

**Review by hubertp — APPROVED** [review]

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/RepoInitialization.scala [hunk 7]
```diff
@@ -62,9 +62,9 @@ class RepoInitialization(
         eventStream.publish(InitializedEvent.SuggestionsRepoInitialized)
       case Failure(ex) =>
         logger.error(
-          "Failed to initialize SQL suggestions repo [{}]. {}",
+          "Failed to initialize SQL suggestions repo [{}].",
           MaskedPath(directoriesConfig.suggestionsDatabaseFile.toPath),
-          ex.getMessage
+          ex
```
Comment (JaroslavTulach) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/RepoInitialization.scala:67]: Good.

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala [hunk 8]
```diff
@@ -238,7 +238,7 @@ object ProgramExecutionSupport {
     val message = error match {
       case _: ThreadInterruptedException =>
         val message = s"Execution of function $itemName interrupted."
-        ctx.executionService.getLogger.log(Level.WARNING, message)
+        ctx.executionService.getLogger.log(Level.FINE, message)
```
Comment (JaroslavTulach) [inline engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala:241]: This is `java.util.logging.Logger`, right? Then the message could be ``"Execution of function {0} interrupted."` and the `itemName` could be sent to the logger as parameter. This applies to other Java logged messages.

**Review by JaroslavTulach — APPROVED** [review]

**Pull request merged**